Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp

```diff
@@ -3561,9 +3561,9 @@ auto OMGIRGenerator::addArrayGet(ExtGCOpType arrayGetKind, uint32_t typeIndex, E
 
 Value* OMGIRGenerator::emitGetArrayPayloadBase(Wasm::StorageType fieldType, Value* arrayref)
 {
-    auto payloadBase = m_currentBlock->appendNew<Value>(m_proc, Add, pointerType(), origin(), arrayref, constant(pointerType(), JSWebAssemblyArray::offsetOfData()));
+    auto payloadBase = m_currentBlock->appendNew<Value>(m_proc, Add, pointerType(), origin(), truncate(arrayref), constant(pointerType(), JSWebAssemblyArray::offsetOfData()));
```

Comment:
If arrayref is a jsvalue, we should have a separate name for the operation that converts it to a JSCell*

Reply:
That's a very good point. Was thinking of renaming `truncate` to something else when merging the file with `WasmOMGIRGenerator.cpp` but can do it earlier. Though in that case, I'd rather submit a separate PR that's just the renames.